genie-agent/src/main/java/com/netflix/genie/agent/execution/services/JobFolderArchiveService.java

```diff
@@ -0,0 +1,39 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.execution.services;
+
+import com.netflix.genie.agent.execution.ExecutionContext;
+import com.netflix.genie.agent.execution.exceptions.ArchiveException;
+
+/**
+ * Service for archiving the job folder.
```

Comment:
Neither comments explain what does it mean to "archive".
Create an archive?
Upload to an archive location?
Both?

Is this specific to the job folder? If not, this service could be generic and have a generic name.

Reply:
Thought I had was - 
-`UploadService` is a service to upload a file/dir to a target location. This service would allow a method to compress and upload. 
-`JobFolderArchiveService` is specific for job folder archival. It will have one implementation for S3 which will know the s3 url based on the env (prod/test) where to upload the job folder. It will have further checks like if the job folder is missing(in case agent fails before a jobFolder is created), it will not use upload service to archive. That way everything related to JobFolderArchival stays in one place
- `ShutDownAction` will use this `JobFolderArchiveService` to do an upload.

We can possibly have another service `CompressService` which will compress a file/dir and return a zipped file. `UploadService` will then only upload the file provided.